azure-nextgen-go-static-website/main.go

```diff
@@ -0,0 +1,179 @@
+					},
+					File: &storage.EncryptionServiceArgs{
+						Enabled: pulumi.Bool(true),
+					},
+				},
+			},
+			Kind: storage.KindStorageV2,
+			NetworkRuleSet: &storage.NetworkRuleSetArgs{
+				Bypass:        storage.BypassAzureServices,
+				DefaultAction: storage.DefaultActionAllow,
+			},
+			ResourceGroupName: resourceGroup.Name,
+			Sku: &storage.SkuArgs{
+				Name: storage.SkuName_Standard_LRS,
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		endpointOrigin := storageAccount.PrimaryEndpoints.Web().ApplyString(func(endpoint string) string {
+			endpoint = strings.ReplaceAll(endpoint, "https://", "")
+			endpoint = strings.ReplaceAll(endpoint, "/", "")
+			return endpoint
+		})
+
+		queryStringCachingBehaviorNotSet := cdn.QueryStringCachingBehaviorNotSet
+		endpoint, err := cdn.NewEndpoint(ctx, "endpoint", &cdn.EndpointArgs{
+			ContentTypesToCompress: pulumi.StringArray{},
+			EndpointName:           cdnEndpointName,
+			IsCompressionEnabled:   pulumi.Bool(false),
+			IsHttpAllowed:          pulumi.Bool(false),
+			IsHttpsAllowed:         pulumi.Bool(true),
+			OriginHostHeader:       endpointOrigin,
+			Origins: cdn.DeepCreatedOriginArray{
+				&cdn.DeepCreatedOriginArgs{
+					HostName:  endpointOrigin,
+					HttpsPort: pulumi.Int(443),
+					Name: pulumi.All(randomSuffix.Result, cdnEndpointName).ApplyString(
+						func(args []interface{}) string {
+							return fmt.Sprintf("%v%v%v", args[1], "-origin-", args[0])
+						}),
+				},
+			},
+			ProfileName:                profile.Name,
+			QueryStringCachingBehavior: &queryStringCachingBehaviorNotSet,
+			ResourceGroupName:          resourceGroup.Name,
+		})
+		if err != nil {
+			return err
+		}
+
+		// Enable static website support
+		staticWebsite, err := storageDefault.NewStorageAccountStaticWebsite(ctx, "staticWebsite", &storageDefault.StorageAccountStaticWebsiteArgs{
+			AccountName:       storageAccount.Name,
+			ResourceGroupName: resourceGroup.Name,
+			IndexDocument:     pulumi.String("index.html"),
+			Error404Document:  pulumi.String("404.html"),
+		})
+		if err != nil {
+			return err
+		}
+
+		// Upload the files
+		blobTypeBlock := storageDefault.BlobTypeBlock
+		_, err = storageDefault.NewBlob(ctx, "index_html", &storageDefault.BlobArgs{
+			BlobName:          pulumi.String("index.html"),
+			ResourceGroupName: resourceGroup.Name,
+			AccountName:       storageAccount.Name,
+			ContainerName:     staticWebsite.ContainerName,
+			Type:              &blobTypeBlock,
+			Source:            pulumi.NewFileAsset("./wwwroot/index.html"),
+			ContentType:       pulumi.String("text/html"),
+		})
+		if err != nil {
+			return err
+		}
+		_, err = storageDefault.NewBlob(ctx, "notfound_html", &storageDefault.BlobArgs{
+			BlobName:          pulumi.String("404.html"),
+			ResourceGroupName: resourceGroup.Name,
+			AccountName:       storageAccount.Name,
+			ContainerName:     staticWebsite.ContainerName,
+			Type:              &blobTypeBlock,
```

Comment:
If we want, we can remove a bunch of optional parameters (like `Type` here). Should I list them all?

Reply:
Yeah, that sounds good. I've just been copying the TS example, so open to suggestions on simplifying.

Reply:
Let's merge it and I'll take a pass when we need to change to azure-native

Reply:
> Let's merge it and I'll take a pass when we need to change to azure-native

I think this example needs to wait on the next provider release. I'm currently getting the following error: `error: Resource type azure-nextgen:storage:StorageAccountStaticWebsite not found` and it seems like some of the APIs aren't available in the current latest release.

Reply:
0.6.1 should have all that's needed... Other languages are fine on it, right?

Reply:
Ok, got it working. I think I was missing a `latest` tag before.